spec/models/column_number_spec.rb

```diff
@@ -72,15 +72,62 @@
         end
       end
 
-      context 'when checkanswers or confirmation page' do
+      context 'when checkanswers page' do
+        # branching fixture 10 CYA page
+        let(:uuid) { 'da2576f9-7ddd-4316-b24b-103708139214' }
         let(:new_column) { 10 }
 
-        # branching fixture 10 CYA and Confirmation pages
-        %w[da2576f9-7ddd-4316-b24b-103708139214 a88694da-8ded-44e7-bc89-e652c1a7f46d].each do |page_uuid|
-          let(:uuid) { page_uuid }
+        context 'when confirmation page is present' do
+          let(:positions) do
+            {
+              # branching fixture 10 Confirmation page
+              'a88694da-8ded-44e7-bc89-e652c1a7f46d' => {
+                column: 11,
+                row: 1
+              }
+            }
+          end
+          let(:latest_column) { 10 }
+
+          it 'returns the latest column' do
+            expect(column_number.number).to eq(latest_column)
+          end
+        end
+
+        context 'when confirmation page is not present' do
```

Comment:
there are at least two contexts for this i think. `latest_column` will return the higher of the `existing_column` or the `new_column`. in your example on line 100 it will always return the `new_column` as an existing checkanswers column does not exist.

so there probably needs to be contexts when there is no existing checkanswers, where there is an existing checkanswers but the number is less than the `new_column` and then where there is an existing checkanswers but the number is greater than the `new_column`.